index.js

```diff
@@ -48,6 +55,201 @@ class ExecutorQueue extends Executor {
+            // eslint-disable-next-line max-len
+            winston.info(`worker[${workerId}] ${job} failure ${queue} ${JSON.stringify(job)} >> ${failure}`));
+        this.multiWorker.on('error', (workerId, queue, job, error) =>
+            winston.error(`worker[${workerId}] error ${queue} ${JSON.stringify(job)} >> ${error}`));
+        this.multiWorker.on('pause', workerId =>
+            winston.info(`worker[${workerId}] paused`));
+
+        // multiWorker emitters
+        this.multiWorker.on('internalError', error =>
+            winston.error(error));
+        this.multiWorker.on('multiWorkerAction', (verb, delay) =>
+            winston.info(`*** checked for worker status: ${verb} (event loop delay: ${delay}ms)`));
+
+        this.scheduler.on('start', () =>
+            winston.info('scheduler started'));
+        this.scheduler.on('end', () =>
+            winston.info('scheduler ended'));
+        this.scheduler.on('master', state =>
+            winston.info(`scheduler became master ${state}`));
+        this.scheduler.on('error', error =>
+            winston.info(`scheduler error >> ${error}`));
+        this.scheduler.on('working_timestamp', timestamp =>
+            winston.info(`scheduler working timestamp ${timestamp}`));
+        this.scheduler.on('transferred_job', (timestamp, job) =>
+            winston.info(`scheduler enqueuing job timestamp  >>  ${JSON.stringify(job)}`));
+
+        process.on('SIGTERM', () => {
+            this.multiWorker.end((error) => {
+                if (error) {
+                    winston.error(`failed to end the worker: ${error}`);
+                }
+
+                this.scheduler.end((err) => {
+                    if (err) {
+                        winston.error(`failed to end the scheduler: ${err}`);
+                        process.exit(128);
+                    }
+                    process.exit(0);
+                });
+            });
+        });
+    }
+
+    /**
+     * Posts a new build event to the API
+     * @method postBuildEvent
+     * @param {Object} config          Configuration
+     * @param {Object} config.pipeline Pipeline of the job
+     * @param {Object} config.job      Job object to create periodic builds for
+     * @return {Promise}
+     */
+    async postBuildEvent({ pipeline, job }) {
+        const jwt = this.tokenGen(pipeline.admins[0], {}, pipeline.scmContext);
+
+        const options = {
+            url: '/events',
+            method: 'POST',
+            headers: {
+                Authorization: `Bearer ${jwt}`,
+                'Content-Type': 'application/json'
+            },
+            body: {
+                pipelineId: pipeline.id,
+                startFrom: job.name
+            }
+        };
+
+        return req(options, (err, response) => {
+            if (err) {
+                return Promise.reject(err);
+            }
+
+            return Promise.resolve(response);
```

Comment:
Please also check `response.statusCode` before resolve, the behavior is kinda weird. It's possible that err object is empty for non 2XX response.

example:
```
request('https://api.screwdriver.cd/ssss', function (err, response) { console.log(response.statusCode); console.log(error); });
```
statusCode is 404 but error object undefined.